lib/pages/chat_list/chat_list_view_builder.dart

```diff
@@ -24,46 +27,109 @@ class ChatListViewBuilder extends StatelessWidget {
         return ValueListenableBuilder(
           valueListenable: controller.selectModeNotifier,
           builder: (context, _, __) {
-            return ValueListenableBuilder(
-              valueListenable: controller.widget.activeRoomIdNotifier,
-              builder: (context, activeRoomId, child) {
-                return ChatListItem(
-                  rooms[index],
-                  key: Key('chat_list_item_${rooms[index].id}'),
-                  isEnableSelectMode: controller.isSelectMode,
-                  onTap: controller.isSelectMode
-                      ? () => controller.toggleSelection(rooms[index].id)
-                      : null,
-                  onSecondaryTap: () => controller.handleContextMenuAction(
-                    context,
-                    rooms[index],
-                  ),
-                  onLongPress: () => controller.onLongPressChatListItem(
-                    rooms[index],
-                  ),
-                  checkBoxWidget: ValueListenableBuilder(
-                    valueListenable: controller.conversationSelectionNotifier,
-                    builder: (context, conversationSelection, __) {
-                      final conversation =
-                          conversationSelection.firstWhereOrNull(
-                        (conversation) =>
-                            conversation.roomId.contains(rooms[index].id),
-                      );
-                      return Checkbox(
-                        value: conversation?.isSelected == true,
-                        onChanged: (_) {
-                          controller.toggleSelection(rooms[index].id);
-                        },
-                      );
-                    },
-                  ),
-                  activeChat: activeRoomId == rooms[index].id,
-                );
-              },
-            );
+            return ChatListViewStyle.responsiveUtils
+                        .isMobileOrTablet(context) &&
+                    !controller.isSelectMode
```

Comment:
use value of valueNotifier here

Reply:
and consider using child

Reply:
> use value of valueNotifier here

no it's not necessary since I use already the valuenotifer selectModeNotifier

Reply:
> Using this pre-built child is entirely optional, but can improve performance significantly in some cases and is therefore a good practice.

IMO it's good for us only for nullable value. In my case there is no interest

Reply:
> > use value of valueNotifier here
> 
> no it's not necessary since I use already the valuenotifer selectModeNotifier

no, but you should use local one, because it's the right way